### PR TITLE
net: openthread: Set state changed callback before openthread start

### PR DESCRIPTION
**Pull request by lmaciejonczyk**

### CODE REVIEW

File: include/net/openthread.h [hunk 1]
```diff
@@ -30,10 +30,16 @@ struct openthread_context {
 	struct pkt_list_elem pkt_list[CONFIG_OPENTHREAD_PKT_LIST_SIZE];
 };
 
+void openthread_set_state_changed_cb(otStateChangedCallback cb);
```
Comment (jukkar) [inline include/net/openthread.h:33]: We are missing documentation to these functions. This is a public header file so we should have documentation to relevant structs/functions in this file. If something is suppose to be private, the `openthread_context` looks like a good candidate, then please mark them using (see examples in other header files in this directory)
```
/** @cond INTERNAL_HIDDEN */
...
/* @endcond */
```

**Review by jukkar — CHANGES_REQUESTED** [review]

**Review by rlubos — COMMENTED** [review]
Looks good overall, but it's a good point that we should start documenting public APIs.

File: include/net/openthread.h [hunk 2]
```diff
@@ -17,27 +28,80 @@
...
 	uint16_t pkt_list_out_idx;
+
+	/** Flag indicates that pkt_list is full */
 	uint8_t pkt_list_full;
+
+	/** Array for storing net_pkt for OpenThread internal usage */
 	struct pkt_list_elem pkt_list[CONFIG_OPENTHREAD_PKT_LIST_SIZE];
 };
+/**
+ * INTERNAL_HIDDEN @endcond
```
Comment (jukkar) [inline include/net/openthread.h:64]: I did not know that this works too, usually I have just placed `/** @endcond */` here

**Review by jukkar — APPROVED** [review]
Thanks! LGTM

File: include/net/openthread.h (hunk 2 above)
Comment (lmaciejonczyk) [reply include/net/openthread.h:64]: It's quite common for instance in `kernel.h`

**Review by rlubos — APPROVED** [review]
Thanks!

**Pull request merged**